### PR TITLE
T17_migrate_home_screen_to_compose

### PR DESCRIPTION
**Pull request by Animansoubi**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -103,13 +103,17 @@ dependencies {
...
+    implementation("io.coil-kt:coil-compose:2.4.0")
+    implementation "androidx.compose.material:material:1.5.3"
     implementation 'androidx.compose.material3:material3'
     implementation 'androidx.compose.ui:ui-tooling-preview'
     debugImplementation 'androidx.compose.ui:ui-tooling'
     implementation 'androidx.activity:activity-compose:1.8.0'
     implementation 'androidx.lifecycle:lifecycle-viewmodel-compose:2.6.2'
     implementation "androidx.compose.material:material-icons-extended:1.5.3"
+    implementation "androidx.navigation:navigation-compose:2.7.4"
+    implementation "androidx.paging:paging-compose:3.3.0-alpha02"
```
Comment (moyarsuccess) [inline app/build.gradle]: Remove this dependency

File: app/src/main/java/com/gathering/android/common/PicassoImageLoader.kt [hunk 2]
```diff
@@ -4,13 +4,12 @@ import android.widget.ImageView
...
 import javax.inject.Inject
 
-
 class PicassoImageLoader @Inject constructor(
     private val baseurl: String,
     private val picasso: Picasso,
 ) : ImageLoader {
 
-    override fun loadImage(imageName: String?, imageView: ImageView) {
+    override fun loadImage(imageName: String?, imageView: ImageView?) {
```
Comment (moyarsuccess) [inline app/src/main/java/com/gathering/android/common/PicassoImageLoader.kt:12]: Why have you defined this image view as nullable?

File: app/src/main/java/com/gathering/android/event/repo/ApiEventRepository.kt [hunk 3]
```diff
@@ -18,15 +19,48 @@ class ApiEventRepository @Inject constructor(
...
             .enqueue(handleGetEventResponse(onResponseReady))
     }
 
     override fun getEvents(
         page: Int,
         onResponseReady: (eventRequest: ResponseState<List<EventModel>>) -> Unit
     ) {
-        eventRemoteService.getMyEvents(pageSize = PAGE_SIZE, pageNumber = page, isMyEvent = false)
+
+//        Thread {
```
Comment (moyarsuccess) [inline app/src/main/java/com/gathering/android/event/repo/ApiEventRepository.kt:31]: Remove this thread

File: app/src/main/java/com/gathering/android/home/HomeScreen.kt [hunk 4]
```diff
@@ -92,8 +160,150 @@ class HomeScreen : Fragment(), HomeNavigator {
...
+                ), Event(
+                    eventId = 2,
+                    eventName = "Mo",
+                    eventHostEmail = "mo@gmail.com",
+                    description = "party2",
+                    photoUrl = "",
+                    latitude = 0.0,
+                    longitude = null
+                )
+            ),
```
Comment (moyarsuccess) [inline app/src/main/java/com/gathering/android/home/HomeScreen.kt:192]: Put name for all these parameters

File: app/src/main/java/com/gathering/android/ui/theme/Screen.kt [hunk 5]
```diff
@@ -0,0 +1,11 @@
+package com.gathering.android.ui.theme
+
+import androidx.annotation.StringRes
+import com.gathering.android.R
+
+sealed class Screen(val route: String, @StringRes val resourceId: Int) {
```
Comment (moyarsuccess) [inline app/src/main/java/com/gathering/android/ui/theme/Screen.kt:6]: Why have you defined this?

File: app/src/main/res/values/MyEvent.xml [hunk 6]
```diff
@@ -0,0 +1,2 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources></resources>
```
Comment (idaosk) [inline app/src/main/res/values/MyEvent.xml:2]: what is this xml for?

File: build.gradle [hunk 7]
```diff
@@ -1,7 +1,7 @@
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 plugins {
-    id("com.android.application") version "8.1.1" apply false
-    id("com.android.library") version "8.1.1" apply false
+    id("com.android.application") version "8.1.0" apply false
+    id("com.android.library") version "8.1.0" apply false
     id 'org.jetbrains.kotlin.android' version '1.9.10' apply false
```
Comment (idaosk) [inline build.gradle:5]: why was the version changed to 8.1.0?

File: app/src/main/java/com/gathering/android/event/Event.kt [hunk 8]
```diff
@@ -16,4 +16,12 @@ data class Event(
     val isMyEvent: Boolean = false,
     val attendees: List<Attendee> = listOf(),
     val liked: Boolean = false,
-) : Serializable
+) : Serializable {
+    companion object {
```
Comment (idaosk) [inline app/src/main/java/com/gathering/android/event/Event.kt:20]: can you please tell me why this was added here? where is this used?

File: app/src/main/java/com/gathering/android/common/Composables.kt [hunk 9]
```diff
@@ -198,5 +199,38 @@ fun ErrorText(error: String) {
 }
 
 
+@Composable
```
Comment (idaosk) [inline app/src/main/java/com/gathering/android/common/Composables.kt:202]: nice job!

File: app/src/main/java/com/gathering/android/event/repo/ApiEventRepository.kt [hunk 10]
```diff
@@ -18,15 +18,15 @@ class ApiEventRepository @Inject constructor(
...
+        eventRemoteService.getMyEvents(pageSize = PAGE_SIZE, pageNumber = page)
             .enqueue(handleGetEventResponse(onResponseReady))
     }
 
     override fun getEvents(
         page: Int,
         onResponseReady: (eventRequest: ResponseState<List<EventModel>>) -> Unit
     ) {
-        eventRemoteService.getMyEvents(pageSize = PAGE_SIZE, pageNumber = page, isMyEvent = false)
+        eventRemoteService.getALlEvents(pageSize = PAGE_SIZE, pageNumber = page)
```
Comment (amirziarati) [inline app/src/main/java/com/gathering/android/event/repo/ApiEventRepository.kt:29]: NIT: L shouldnt be capital letter.

**Review by moyarsuccess — APPROVED** [review]

File: build.gradle (hunk 7 above)
Comment (idaosk) [reply build.gradle:5]: @Animansoubi

**Pull request merged**